apps/homescreen/js/app.js

```diff
@@ -918,6 +918,21 @@
 
       // Add apps installed after startup
       case 'install':
+        // Check if the app already exists, and if so, update it.
+        // This happens when reinstalling an app via WebIDE.
+        var existing = false;
+        for (child of this.icons.children) {
+          icon = child.firstElementChild;
+          if (icon.app && icon.app.manifestURL === e.application.manifestURL) {
+            icon.app = e.application;
+            icon.refresh();
+            existing = true;
```

Comment:
Just spoke on IRC, but to add here for record - I don't break here in case the app has multiple entry points. Note that this still doesn't handle entry points quite correctly, but I heard that they'd be disappearing at some point, so I'm loathe to complicate the code for something that isn't really used much. We can deal with the issues if they get raised.